motor/core.py

```diff
@@ -1120,11 +1119,19 @@ def __init__(self, cursor, collection):
     def __aiter__(self):
         return self
 
-    async def __anext__(self):
-        # An optimization: skip the "await" if possible.
-        if self._buffer_size() or await self.fetch_next:
-            return self.next_object()
-        raise StopAsyncIteration()
+    async def next(self):
+        """Advance the cursor.
+
+        .. versionadded:: 2.2
+        """
+        if not self.alive:
```

Comment:
We have to explicitly check this condition here (instead of relying on the same exception to be raised by `_get_more` and catching it here) because using `except InvalidOperation: raise StopAsyncIteration` actually masks command errors that occur during the `getMore`.

Reply:
Ack. Minor nit: We could reduce this code to:
```python
        if self.alive and (self._buffer_size() or await self._get_more()):
            return next(self.delegate)
        raise StopAsyncIteration
```

Reply:
Nice. Done.